### PR TITLE
Add Derive FieldType support for newtypes

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: examples/getting_started_uuid/.butane/migrations/current/types.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"CT:Blog":{"KnownId":{"Ty":"Json"}},"CT:BlogId":{"KnownId":{"Ty":"Blob"}},"CT:PostId":{"KnownId":{"Ty":"Blob"}}}
```
Comment (jayvdb) [inline examples/getting_started_uuid/.butane/migrations/current/types.json:1]: The Id fields are Blob instead of Json

Comment (jayvdb) [reply examples/getting_started_uuid/.butane/migrations/current/types.json]: Ignore the
> "CT:Blog":{"KnownId":{"Ty":"Json"}}

That was a stale entry.

File: examples/getting_started_uuid/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# butane `getting_started_uuid` example
```
Comment (Electron100) [inline examples/getting_started_uuid/README.md:1]: I'm not quite following why we need a whole separate example that's mostly a copy of the other one except it makes use of deriving a FieldType? Why not one of
1. Just a test using this functionality
2. Extend one of the models in the getting_started example with a new field showcasing this
3. A new wholly separate (and probably smaller, without several separate commands) example showcasing this.

File: butane_codegen/Cargo.toml [hunk 3]
```diff
@@ -17,6 +17,7 @@ uuid = ["butane_core/uuid"]
 butane_core = { workspace = true }
 proc-macro2 = { workspace = true }
 quote = { workspace = true }
+serde_variant = "0.1"
```
[the remark this answers is not in the document]
Comment (Electron100) [reply butane_codegen/Cargo.toml:20]: Is this actually even used?

File: butane_codegen/src/lib.rs [hunk 4]
```diff
@@ -240,12 +240,71 @@ pub fn derive_field_type(input: TokenStream) -> TokenStream {
...
+            fields: syn::Fields::Unit { .. },
+            ..
+        }) => derive_field_type_with_json(ident),
         syn::Data::Enum(data_enum) => derive_field_type_for_enum(ident, data_enum),
         syn::Data::Union(_) => derive_field_type_with_json(ident),
     }
 }
 
+fn derive_field_type_for_newtype(ident: &Ident, sqltype: SqlType) -> TokenStream {
+    let sqltype_name = serde_variant::to_variant_name(&sqltype).unwrap();
```
Comment (jayvdb) [inline butane_codegen/src/lib.rs:271]: `serde_variant` used here

File: butane_codegen/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jayvdb) [reply butane_codegen/Cargo.toml:20]: Sorry I didnt notice the license ; it isn't hard to replace

File: examples/getting_started_uuid/src/models.rs [hunk 5]
```diff
@@ -0,0 +1,86 @@
+//! Models for the getting_started example.
```
Comment (jayvdb) [inline examples/getting_started_uuid/src/models.rs:1]: replace "getting_started" with whatever name we use

File: examples/getting_started_uuid/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+//! Common helpers for the getting_started example CLI.
```
Comment (jayvdb) [inline examples/getting_started_uuid/src/lib.rs:1]: replace "getting_started" with whatever name we use

File: butane_codegen/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jayvdb) [reply butane_codegen/Cargo.toml]: Replaced with strum.

File: examples/getting_started_uuid/README.md (hunk 2 above)
Comment (Electron100) [reply examples/getting_started_uuid/README.md:1]: Your points are well taken. I like this suggestion:

> I could change this example to be "newtype_uuid" ; happy to drop all of the src/bin.

I think that preserves the benefits of this being an example while being clearer on its purpose and reducing duplication

File: butane_codegen/Cargo.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jayvdb) [reply butane_codegen/Cargo.toml]: https://github.com/d-e-s-o/serde_variant/issues/4 was resolved - I'm using serde_variant again as it is much lighter than strum.

**Review by Electron100 — APPROVED** [review]

**Pull request merged**